### PR TITLE
Rewrite: Add help and example content for CLI module

### PR DESCRIPTION
**Pull request by m90**
This is a first pass at adding some content and structure to the output of `mochify --help`. Right now, this is relatively wordy (maybe a little too wordy) as it's the only documentation around, I think it should be possible to reduce the amount of inline docs over time (moving it to `@mochify/mochify`).

### CODE REVIEW

File: cli/index.js [hunk 1]
```diff
@@ -6,31 +6,82 @@ const { hideBin } = require('yargs/helpers');
...
+    describe:
+      'Free form options to pass to the driver in use. Pass an arbitrary number of options using `--driver-option.foo 1 --driver-option.bar 2`. Refer to the documentation of the driver in use for available options.'
   })
   .option('reporter', {
     alias: 'R',
-    type: 'string'
+    type: 'string',
+    group: 'Options:',
+    describe:
+      'The Mocha reporter to use. Right now, only reporters that are included with Mocha itself can be used.'
```
Comment (m90) [inline cli/index.js:53]: This might not be correct.

File: cli/index.js [hunk 2]
```diff
@@ -6,31 +6,82 @@ const { hideBin } = require('yargs/helpers');
...
+        );
+    }
+  )
   .option('config', {
-    type: 'string'
+    alias: 'C',
+    type: 'string',
+    group: 'Options:',
+    describe:
+      'The config file to use. In case `package.json` is given, the configuration is expected to be stored in a top-level "mochify" key. In case an option is present in both the config file and as a CLI flag, the flag takes precedence. Refer to the documentation of `@mochify/mochify` for available configuration options.'
```
Comment (m90) [inline cli/index.js:33]: Any opinions on handling long strings like this in regards to line length?

File: cli/index.js [hunk 3]
```diff
@@ -6,31 +6,82 @@ const { hideBin } = require('yargs/helpers');
...
+    'Run Mocha tests using Mochify.',
+    (local) => {
+      local
+        .example(
+          '$0 --driver puppeteer --bundle browserify "./src/**/*.test.js" ',
+          'Bundle all files matching the given spec using browserify and run them using @mochify/driver-puppeteer.'
+        )
+        .example(
+          '$0 --config package.json "./src/**/*.test.js" ',
+          'Run all tests matching the given spec using the configuration from package.json\'s "mochify" key.'
```
Comment (mantoni) [inline cli/index.js:20]: The `mochify` key in `package.json` is picked up automatically. Specifying it like this would be superfluous.

File: cli/index.js [hunk 4]
```diff
@@ -6,31 +6,82 @@ const { hideBin } = require('yargs/helpers');
 const { mochify } = require('@mochify/mochify');
 
 const opts = yargs(hideBin(process.argv))
+  .usage(
+    '$0 [options] <spec...>',
+    'Run Mocha tests using Mochify.',
```
Comment (mantoni) [inline cli/index.js:11]: "Run Mocha tests in real browsers."?

File: cli/index.js (hunk 2 above)
Comment (mantoni) [reply cli/index.js:33]: What happens if you use a multiline string? Does it screw up the layout?

Otherwise we can just concatenate strings with `+`.

File: cli/index.js (hunk 1 above)
Comment (mantoni) [reply cli/index.js:53]: True. This might actually work.

**Review by mantoni — COMMENTED** [review]
Yeah, I agree that this is probably too wordy. Comparing it to the `mocha -h` output, they only have very short descriptions. I like the content you've written though. We could put it in the readme (and therefore on the website).

What I quite like about the mocha help output is that they link to the repo and the documentation website.

File: cli/index.js (hunk 3 above)
Comment (m90) [reply cli/index.js:20]: Oh that's interesting to know.

File: cli/index.js [hunk 5]
```diff
@@ -8,80 +8,82 @@ const { mochify } = require('@mochify/mochify');
...
-          '$0 --config package.json "./src/**/*.test.js" ',
-          'Run all tests matching the given spec using the configuration from package.json\'s "mochify" key.'
+          '$0 --esm --reporter dot --driver puppeteer "./src/**/*.test.js" ',
+          'Run all tests matching the given spec as ES modules in puppeteer and use the "dot" reporter for output.'
         )
         .example(
-          '$0 --esm --reporter dot "./src/**/*.test.js" ',
-          'Run all tests matching the given spec as ES modules and use the "dot" reporter for output.'
+          '$0 "./src/**/*.test.js" ',
+          'Run all tests matching the given spec using the configuration from package.json'
```
Comment (mantoni) [inline cli/index.js:24]: If there is a `mochify` key in `package.json`. cosmiconfig tries various options in a specific order. Maybe we should just document that with a link to cosmiconfig?

See https://github.com/davidtheclark/cosmiconfig#explorersearch

File: cli/index.js [hunk 6]
```diff
@@ -8,80 +8,82 @@ const { mochify } = require('@mochify/mochify');
...
         );
     }
   )
   .option('config', {
     alias: 'C',
     type: 'string',
     group: 'Options:',
-    describe:
-      'The config file to use. In case `package.json` is given, the configuration is expected to be stored in a top-level "mochify" key. In case an option is present in both the config file and as a CLI flag, the flag takes precedence. Refer to the documentation of `@mochify/mochify` for available configuration options.'
+    describe: 'The config file to use (defaults to "package.json")'
```
Comment (mantoni) [inline cli/index.js:40]: See above comment. There's no default.

File: cli/index.js (hunk 5 above)
Comment (m90) [reply cli/index.js:24]: Ah ok, that's not entirely obvious tbh (I remember seeing a list of filenames being removed in a commit once and thought this meant there is no default lookup happening, but apparently that's not the case). Will adjust this accordingly.

Comment (mantoni) [reply cli/index.js:24]: Yes, I removed the file list because I discovered that I replicated the cosmiconfig defaults 😆

File: cli/index.js [hunk 7]
```diff
@@ -6,31 +6,83 @@ const { hideBin } = require('yargs/helpers');
...
   .option('server-option', {
-    type: 'object'
+    type: 'object',
+    group: 'Options:',
+    describe: 'Pass options to the server (requires --serve or --esm)'
   })
+  .updateStrings({
+    'Options:': 'Other:'
+  })
+  .wrap(process.stdout.columns ? Math.min(process.stdout.columns, 80) : 80)
```
Comment (m90) [inline cli/index.js:85]: This could also use higher threshold than '80` in case `process.stdout.columns` is given? Probably _very_ subjective.

Comment (mantoni) [reply cli/index.js:85]: Behaving the same as mocha is generally a good approach.

**Review by mantoni — APPROVED** [review]
👍😎
